lib/utils.js

```diff
@@ -67,6 +68,21 @@ function addArrayToSet(array, set, valueExtractor) {
   return set
 }
 
+// merge the given definition onto the base definition. Be careful to handle various arrays
+// correctly by finding matching entried and merging them as appropriate.
+function merge(base, newDefinition) {
```

Comment:
we may want to update the name of this or make the implementation more generic.

Maybe we could leave it called `merge()` and update to check all the props for `IsArray()` and merge everything

or maybe we could leave the implementation and rename it to `mergeFiles()` or something like that?

Reply:
renamed to mergeDefinitions